app_https_client.py

```diff
@@ -141,38 +161,50 @@ def _sendGPSData():
             time.sleep(1)
     except Exception as e:
         print(e)
-        print("_sendGPSData failed. Retrying in", retry_interval, "second(s)")
-        time.sleep(retry_interval)
+        print("_sendGPSData failed. Initiating graceful exit.")
+        failure_queue.put(e)
 
 
 def start_jobs():
     print("Starting jobs...")
     global jobs
     global rawFrames
     global encodedFrames
+    global failure_queue
+
     pipelineProcess = multiprocessing.Process(target=_pipelineFunc)
     encodingProcess = multiprocessing.Process(target=_encodingFunc)
     videoSendingProcess = multiprocessing.Process(target=_sendVideoData)
     gpsSendingProcess = multiprocessing.Process(target=_sendGPSData)
-    jobs = [pipelineProcess, encodingProcess, videoSendingProcess, gpsSendingProcess]
+    jobs = [pipelineProcess, encodingProcess, videoSendingProcess]
```

Comment:
Maybe a better way would be a command line arg --enableGPS which is false by default.

Reply:
Agreed and fixed.